### PR TITLE
Fix minor memory leaks

### PR DESCRIPTION
**Pull request by williameveretteggplant**
A leak of NSColor objects in NSBox, and GSToolTip objects in NSView

### CODE REVIEW

**Comment by fredkiefer** [comment]
I fully understand the NSBox change. For the tool tips change in NSView I am sure that it won't do any harm but at the moment I don't see why this change would be needed? A view start off without tool tips and when we add one we also set the flag. We even don't clear the flag when we remove the tool tips. That way it looks to me as the flag should always be set when there is a tool tip to remove. Why do you think this isn't the case?

**Comment by williameveretteggplant** [comment]
There appears to be a line number mixup. From what I can see that line number is part of the writePDFInsideRect:toPasteboard: method and doesn't have anything to do with Tool Tips. Could you clarify what method you are talking about?

**Comment by fredkiefer** [comment]
No, it is worse. I did not mix up the line number but the file name. I wanted to refer to NSTextView.m not NSView.m. Sorry for the confusion.

File: Source/NSView.m [hunk 1]
```diff
@@ -750,10 +750,9 @@ - (void) dealloc
   TEST_RELEASE(_frameMatrix);
   TEST_RELEASE(_boundsMatrix);
   TEST_RELEASE(_sub_views);
-  if (_rFlags.has_tooltips != 0)
-    {
-      [GSToolTips removeTipsForView: self];
-    }
+
+  [GSToolTips removeTipsForView: self];
```
Comment (fredkiefer) [inline Source/NSView.m:754]: Did you test, whether you still need this change? It won't do any harm, maybe slow things a bit down. But what we really should know is whether we addressed the underlying issue.

File: Source/NSView.m [hunk 2]
```diff
@@ -750,10 +750,11 @@ - (void) dealloc
   TEST_RELEASE(_frameMatrix);
   TEST_RELEASE(_boundsMatrix);
   TEST_RELEASE(_sub_views);
+
   if (_rFlags.has_tooltips != 0)
-    {
-      [GSToolTips removeTipsForView: self];
-    }
+  {
```
Comment (fredkiefer) [inline Source/NSView.m:755]: Looks like the indentation got messed up. Could you please fix this?

**Review by fredkiefer — APPROVED** [review]

**Pull request merged**